tests/load/filesystem/test_object_store_rs_credentials.py

```diff
@@ -78,66 +85,91 @@ def test_azure_object_store_rs_credentials() -> None:
     assert can_connect(AZ_BUCKET, creds.to_object_store_rs_credentials())
 
 
-@pytest.mark.skipif(
-    "s3" not in ALL_FILESYSTEM_DRIVERS, reason="`s3` not in `ALL_FILESYSTEM_DRIVERS`"
+@pytest.mark.parametrize(
+    "driver", [driver for driver in ALL_FILESYSTEM_DRIVERS if driver in ("s3", "r2")]
 )
-def test_aws_object_store_rs_credentials() -> None:
+def test_aws_object_store_rs_credentials(driver: str) -> None:
     creds: AwsCredentialsWithoutDefaults
 
+    fs_creds = FS_CREDS
+    if driver == "r2":
+        fs_creds = R2_BUCKET_CONFIG["credentials"]  # type: ignore[assignment]
+
+    # AwsCredentialsWithoutDefaults: no user-provided session token
+    creds = AwsCredentialsWithoutDefaults(
+        aws_access_key_id=fs_creds["aws_access_key_id"],
+        aws_secret_access_key=fs_creds["aws_secret_access_key"],
+        region_name=fs_creds.get("region_name"),
+        endpoint_url=fs_creds.get("endpoint_url"),
+    )
+    assert creds.aws_session_token is None
+    object_store_rs_creds = creds.to_object_store_rs_credentials()
+    assert "aws_session_token" not in object_store_rs_creds  # no auto-generated token
+    assert can_connect(AWS_BUCKET, object_store_rs_creds)
+
     # AwsCredentials: no user-provided session token
     creds = AwsCredentials(
-        aws_access_key_id=FS_CREDS["aws_access_key_id"],
-        aws_secret_access_key=FS_CREDS["aws_secret_access_key"],
-        # region_name must be configured in order for data lake to work
-        region_name=FS_CREDS["region_name"],
+        aws_access_key_id=fs_creds["aws_access_key_id"],
+        aws_secret_access_key=fs_creds["aws_secret_access_key"],
+        region_name=fs_creds.get("region_name"),
+        endpoint_url=fs_creds.get("endpoint_url"),
     )
     assert creds.aws_session_token is None
     object_store_rs_creds = creds.to_object_store_rs_credentials()
-    assert object_store_rs_creds["aws_session_token"] is not None  # auto-generated token
+    assert "aws_session_token" not in object_store_rs_creds  # no auto-generated token
     assert can_connect(AWS_BUCKET, object_store_rs_creds)
 
+    # exception should be raised if both `endpoint_url` and `region_name` are
+    # not provided
+    with pytest.raises(ObjectStoreRsCredentialsException):
+        AwsCredentials(
+            aws_access_key_id=fs_creds["aws_access_key_id"],
+            aws_secret_access_key=fs_creds["aws_secret_access_key"],
+        ).to_object_store_rs_credentials()
+
+    if "endpoint_url" in object_store_rs_creds:
+        # TODO: make sure this case is tested on GitHub CI, e.g. by adding
```

Comment:
are you planning to do this? i think the dremio tests already use minio. Not sure wether it is needed though...